release_notes.md

```diff
@@ -2,6 +2,7 @@
 <!-- Please add your release notes in the following format:
 - My change description (#PR)
 -->
+- Added a feature flag to opt out of the default behavior where the host sets the environment name to `Development` when running in debug mode. To disable the behavior, set the app setting: `AzureWebJobsFeatureFlags` to `DisableDevModeInDebug`
```

Comment:
Should the setting here specifically mention the Dev *Environment*? Don't other places check `InDebugMode` as well? Or is that it?

Reply:
Other places use the debug mode flag to apply logic to logging, but that's not a dev mode. I'm open to other suggestions for this name.

Reply:
I also think there's code that changes things like the lease expirations and queue polling delay, etc...

Reply:
Yes, that applies to the debug and local dev as well.